### PR TITLE
Task_DEV_4

### PR DESCRIPTION
**Pull request by PolinaYeroshevich**
Created a class of rectangles and methods for finding the perimeter and area, as well as checking whether it is a square.

### CODE REVIEW

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            }
+            else
+            {
+                Console.WriteLine("It's not a square");
+            }
+            return isNotSquare;
+        }
+    }
+}
+
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Unnecessary line.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            double width = 7;
+            rectangle.CreateLengthAndWidth(length,width);
+            double rectangleArea = rectangle.FindArea();
+            Console.WriteLine($"The area of a rectangle is {rectangleArea}");
+            double rectanglePerimetr = rectangle.FindPerimetr();
+            Console.WriteLine($"The perimetr of a rectangle is {rectanglePerimetr}");
+            bool isNotSquare = rectangle.CheckForSquare();
+        }
+    }
+}
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Please add blank line to the end of file.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+
+
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Unnecessary line.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Task_DEV_4
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Rectangle rectangle = new Rectangle ();
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Unnecessary whitespace.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            double length = 4;
+            double width = 7;
+            rectangle.CreateLengthAndWidth(length,width);
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Where is whitespace after comma?

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        }       
+
+        public Rectangle(double length, double width)
+        {
+            this.length = length;
+            this.width = width;
+        }
+
+        public void CreateLengthAndWidth(double length, double width)
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Bad method name.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            this.length = length;
+            this.width = width;
+        }
+
+        public double FindArea()
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Bad method name.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            return length * width;
+        }
+
+        public double FindPerimetr()
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Bad method name.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            return (length + width) * 2;
+        }
+        
+        public bool CheckForSquare ()
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Unnecessary whitespace.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            bool isNotSquare = true;
+            if (length == width)
+            {
+                isNotSquare = false;
+                Console.WriteLine("It's a square");
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Single responsibility violation.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+            }
+            else
+            {
+                Console.WriteLine("It's not a square");
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Single responsibility violation.

**Review by VadimBelkin — CHANGES_REQUESTED** [review]
Please be more attentive to the  code style. We are learning to write production code. It is important.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 1 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 2 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 3 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 4 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 5 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 6 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 7 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 8 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 9 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 10 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 11 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+        }
+
+        public Rectangle(double length, double width)
+        {
+            this.length = length;
+            this.width = width;
+        }
+
+        public void SettingFieldValues(double length, double width)
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: The first word in the name of the method should be verb. Smth like `SetRectangleParameters` would be better.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            this.length = length;
+            this.width = width;
+        }
+
+        public double CalculateTheArea()
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Usually, articles are not written in the name of methods. Just `CalculateArea`

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            return length * width;
+        }
+
+        public double CalculateThePerimeter()
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Usually, articles are not written in the name of methods. Just `CalculatePerimeter`

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            return (length + width) * 2;
+        }
+
+        public bool CheckForSquare()
+        {
+            bool isNotSquare = true;
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Better not to use inverse logic. Direct logic looks clearer. Instead of  `isNotSquare`, use `isSquare`.

Comment (VadimBelkin) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: This method can look like:
```
public bool IsSquare()
{
    return length == width;
}
```

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            Rectangle rectangle = new Rectangle();
+            double length = 4;
+            double width = 7;
+            rectangle.SettingFieldValues(length, width);
+            double rectangleArea = rectangle.CalculateTheArea();
+            Console.WriteLine($"The area of a rectangle is {rectangleArea}");
+            double rectanglePerimetr = rectangle.CalculateThePerimeter();
+            Console.WriteLine($"The perimetr of a rectangle is {rectanglePerimetr}");
+            bool isNotSquare = rectangle.CheckForSquare();
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Better not to use inverse logic. Direct logic looks clearer. Instead of `isNotSquare`, use `isSquare`.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+            if (isNotSquare)
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: It is possible to do smth like:
`if(rectangle.CheckForSquare())`

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 18]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace Task_DEV_4
+{
+    class Program
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: If your file is named like EntryPoint it would be better to rename `class Program` to` class EntryPoint`

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 12 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 13 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 14 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 15 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 16 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 18 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 17 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        static void Main(string[] args)
+        {
+            Rectangle rectangle = new Rectangle();
+            double length = 4;
+            double width = 7;
+            rectangle.SetRectangleParameters(length, width);
+            double rectangleArea = rectangle.CalculateArea();
+            Console.WriteLine($"The area of a rectangle is {rectangleArea}");
+            double rectanglePerimetr = rectangle.CalculatePerimeter();
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: rectanglePerimetEr

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+            Console.WriteLine($"The perimetr of a rectangle is {rectanglePerimetr}");
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: perimetEr

File: Task_DEV_4/Task_DEV_4/Rectangle.cs [hunk 21]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/Rectangle.cs]: Where do we use this using?

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 19 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 20 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/Rectangle.cs (hunk 21 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/Rectangle.cs]: Done.

**Review by VadimBelkin — APPROVED** [review]

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+            double length = 4;
+            double width = 7;
+            rectangle.SetRectangleParameters(length, width);
+            double rectangleArea = rectangle.CalculateArea();
+            Console.WriteLine($"The area of a rectangle is {rectangleArea}");
+            double rectanglePerimeter = rectangle.CalculatePerimeter();
+            Console.WriteLine($"The perimetr of a rectangle is {rectanglePerimeter}");
+            if (rectangle.IsSquare())
+            {
+                Console.WriteLine("This is rectangle");
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Wrong message.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+            }
+            else
+            {
+                Console.WriteLine("This is square");
```
Comment (VadimBelkin) [inline Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Wrong message.

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 22 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

File: Task_DEV_4/Task_DEV_4/EntryPoint.cs (hunk 23 above)
Comment (PolinaYeroshevich) [reply Task_DEV_4/Task_DEV_4/EntryPoint.cs]: Done.

**Review by VadimBelkin — APPROVED** [review]
